lib/bundle.js

```diff
@@ -555,15 +714,39 @@ function generateComponentsObject (documentContext, rootContent, refTypeResolver
             isMissingNode = documentContext.missing.find((missingNode) => {
               return missingNode.path === nodeRef;
             });
+
           if (isMissingNode) {
             refData.nodeContent = refData.node;
             refData.local = false;
           }
           else if (!refData) {
             return;
           }
+          else if (isExtRef(property, '$ref') && !isExtURLRef(property, '$ref')) {
```

Comment:
Let's keep the existing condition logic for `else` statement as is and use `(!isExtRef(property, '$ref') && isExtURLRef(property, '$ref')` as condition and add new logic here.

Reason behind it is as logic for HTTP is very specific and to handle future scenarios better as existing logic is more generic.